### PR TITLE
Add better compiler error message if C++17 compatibility isn't found

### LINKED ISSUE #20483: Need better error if C++17 support is missing
## Reason
```
error: expected ';' after expression
  std::scoped_lock lock(NodalNormalsPreprocessor::_nodal_normals_mutex);
```

Is simply too cryptic for Joe User. Conda dependencies frequently get locked up and do not update to the latest version when running `conda update --all`. Something is happening, and they think the update was successful, but they are not all the way up to date. 

## Design
A simple check of the `CXXFLAGS` would go a long way. If the flags still contain `--std=c++14` a clear error message sould be printed to the screen, prompting the user to conda env remove and conda create teh moose env again.

## Impact
Avoid user confusion. @cticenhour

### PR DESCRIPTION
**Pull request by cticenhour**
Thanks for the `static_assert` suggestion @loganharbour!

Closes #20483

### CODE REVIEW

File: framework/include/base/Moose.h [hunk 1]
```diff
@@ -38,12 +38,18 @@ void MooseMatView(SparseMatrix<Number> & mat);
...
-#ifndef LIBMESH_HAVE_CXX11
-#error MOOSE requires a C++11 compatible compiler (GCC >= 4.8.4, Clang >= 3.5.1). Please update your compiler and try again.
-#endif
+namespace Moose
+{
+static_assert(__cplusplus >= 201703L,
+              "MOOSE requires a C++17 compatible compiler (GCC >= 7.5.0, Clang >= 5.0.2). Please "
+              "update your compiler or, if compatible, add '-std=c++17' to your compiler flags "
+              "and try again. If using the MOOSE conda package, please create a new MOOSE "
+              "environment to ensure an update to the newest available package.");
```
Comment (GiudGiud) [inline framework/include/base/Moose.h:51]: I dont think conda users need a new environment. They just need to update, and if unsuccessful, create a new environment. Yaqi, myself and at least one other only had to update

Comment (dschwen) [reply framework/include/base/Moose.h:51]: To many users have needed to wipe their env

Comment (dschwen) [reply framework/include/base/Moose.h:51]: How about providing a script to wipe and rebuild the env?

Comment (cticenhour) [reply framework/include/base/Moose.h]: @GiudGiud I updated the message to include both options, with update being priority. 

@dschwen Not a bad idea - though we cannot know beforehand what name they chose for their moose environment, so I would prefer pointing them to some instructions they can tailor if needed. We have website instructions that outline how to wipe and get started again, so I added that URL to the message to make it easier.

File: framework/include/base/Moose.h [hunk 2]
```diff
@@ -38,12 +38,21 @@ void MooseMatView(SparseMatrix<Number> & mat);
...
-#ifndef LIBMESH_HAVE_CXX11
-#error MOOSE requires a C++11 compatible compiler (GCC >= 4.8.4, Clang >= 3.5.1). Please update your compiler and try again.
-#endif
+namespace Moose
+{
+static_assert(__cplusplus >= 201703L,
+              "MOOSE requires a C++17 compatible compiler (GCC >= 7.5.0, Clang >= 5.0.2). Please "
+              "update your compiler or, if compatible, add '-std=c++17' to your compiler flags "
+              "and try again. If using the MOOSE conda package, please attempt a MOOSE environment "
+              "update (using `conda update --all`). If this update is not successful, please "
```
Comment (GiudGiud) [inline framework/include/base/Moose.h:51]: mamba?

Comment (cticenhour) [reply framework/include/base/Moose.h]: Nice catch - thanks

**Review by GiudGiud — APPROVED** [review]

**Review by loganharbour — APPROVED** [review]

**Review by dschwen — APPROVED** [review]

**Pull request merged**